### PR TITLE
Jetpack: improve the performance of the disconnect dialog

### PR DESCRIPTION
**Pull request by jim-coffey**
The plugins page load is slow with Jetpack activated - it has been suggested that a lot of time is spent on the PHP side initialising state for the React disconnect modal.

#### Changes proposed in this Pull Request:

1. This PR stops using the main Jetpack app's PHP initial state helper, and instead fetches the data it needs async from the React side.
2. Reduced the size of the redux store and state required by the app - saves 100+Kb raw, 10Kb Gzipped

This is aimed all at improving the performance and load time of the wp-admin plugins page - no longer making all the requests that were being made for the full blown React initial state.

The loading of data is not necessary to allow the user to disconnect.

Slow user data fetching will prevent tracking events on the dialog clicks, but that usually won't happen and I'm not sure we should make the user wait before they can disconnect in those rare cases where fetch requests are slow.

Also, slow requests for connected plugins, or site benefits result in the following fallback display:
fall_back

As a result, I don't think it's worth trying to show any busy text or spinners. The data loads in the background before opening the modal, and the data isn't critical to allow the user to disconnect, and the fallback display looks good anyway.

#### Jetpack product discussion

p1HpG7-ere-p2#comment-52144

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

Performance testing will be done separately to ensure this PR is worthwhile - but basic navigation/refreshing of the admin Plugins page should certainly not take any longer to load than usual, even if there is no immediately obvious speed increase.

We should perform some regression testing to ensure the disconnect modal works as it did previously from the Plugins page.

- the modal from Jetpack dashboard Managing site connection **should remain unchanged.**
- note that the disconnect modal shown on the plugins page disconnects immediately and refreshes the page without showing the reason for disconnect, and thank you steps.
- the plugins page disconnect modal may show placeholders while it waits for data to be fetched (emulate a slow network if necessary)
- otherwise, the plugins page disconnect modal should behave as before

### CODE REVIEW

**Review by travisw — DISMISSED** [review]
This looks good to me and I think is ready for Crew to review. I left a couple comments/questions.

Functionality-wise, it all still works well.
FWIW, I ran Lighthouse tests and do see a nice performance boost. I used a Pressable site.

### Before
Screen Shot 2022-02-18 at 12 07 06 PM
Screen Shot 2022-02-18 at 12 06 34 PM

### After
Screen Shot 2022-02-18 at 12 10 29 PM
Screen Shot 2022-02-18 at 12 10 38 PM

File: projects/plugins/jetpack/_inc/client/state/plugins-store.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
```
[the remark this answers is not in the document]
Comment (jim-coffey) [reply projects/plugins/jetpack/_inc/client/state/plugins-store.js:1]: Fair point :thumbsup:

File: projects/plugins/jetpack/_inc/client/state/licensing/reducer.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ import {
  * @param {object} action
```
[the remark this answers is not in the document]
Comment (jim-coffey) [reply projects/plugins/jetpack/_inc/client/state/licensing/reducer.js:20]: Agreed, I actually tried to remove these, but the `jetpack build plugins/jetpack` build was still failing - I might try just clearing things down and re-building from scratch.

**Review by jeherve — CHANGES_REQUESTED** [review]
Could you rebase one last time so we can review and merge?

File: projects/plugins/jetpack/_inc/client/state/plugins-store.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jim-coffey) [reply projects/plugins/jetpack/_inc/client/state/plugins-store.js]: Done

File: projects/plugins/jetpack/_inc/client/state/licensing/reducer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jim-coffey) [reply projects/plugins/jetpack/_inc/client/state/licensing/reducer.js]: Re-built and removed these changes.

**Comment by jim-coffey** [comment]
> Could you rebase one last time so we can review and merge?

Responded to small review comments, rebased, re-tested.

**Review by jeherve — APPROVED** [review]
This looks good, this should be ready to merge! I would only recommend adding some documentation, but this can be done in a follow-up PR.


*******


Could you add a section in https://github.com/Automattic/jetpack/blob/update/disconnect-dialog-initial-state/projects/plugins/jetpack/_inc/client/README.md about the new store? I worry that it may get confusing in the future, when we won't necessarily remember why this was built that way.

**Comment by jim-coffey** [comment]
r240480-wpcom

**Pull request merged**